source/patches/gamepatches.c

```diff
@@ -243,6 +250,231 @@ void PrivateServerPatcher(void *addr, u32 len, u8 privateServer, const char *ser
 		domainpatcher(addr, len, serverAddr);
 }
 
+static inline int GetOpcode(unsigned int* instructionAddr)
+{
+	return ((*instructionAddr >> 26) & 0x3f);
+}
+
+static inline int GetImmediateDataVal(unsigned int* instructionAddr)
+{
+	return (*instructionAddr & 0xffff);
+}
+
+static inline int GetLoadTargetReg(unsigned int* instructionAddr)
+{
+	return (int)((*instructionAddr >> 21) & 0x1f);
+}
+
+static inline int GetComparisonTargetReg(unsigned int* instructionAddr)
+{
+	return (int)((*instructionAddr >> 16) & 0x1f);
+}
+
+
+s8 do_new_wiimmfi_nonMKWii(void *addr, u32 len) {
+
+		// As of February 2021, Wiimmfi requires a special Wiimmfi patcher 
+		// update which does a bit more than just patch the server adresses. 
+		// This function is being called by apploader.c, right before 
+		// jumping to the entry point (only for non-MKWii games on Wiimmfi), 
+		// and applies all the necessary security fixes to the game. 
+
+		// This function has been implemented by Leseratte. Please don't
+		// try to modify it without speaking to the Wiimmfi team because
+		// doing so could have unintended side effects. 
+
+		// This function MUST not run for MKWii, that would break stuff. 
+
+		int hasGT2Error = 0;
+        char gt2locator[] = { 0x38, 0x61, 0x00, 0x08, 0x38, 0xA0, 0x00, 0x14};        
```

Comment:
I would like to note that the indentation is not consistent (space versus tab).

Reply:
Oops. I'll update that later today.

Reply:
I had a little free time, so I formatted the whole file.